lightning/src/ln/channelmanager.rs

```diff
@@ -681,23 +674,25 @@ pub type SimpleRefChannelManager<'a, 'b, 'c, 'd, 'e, M, T, F, L> = ChannelManage
 //  |
 //  |__`forward_htlcs`
 //  |
-//  |__`channel_state`
+//  |__`pending_background_events`
```

Comment:
I'd just like to ask if `process_background_events` through `timer_tick_occurred` can be called by a thread while another thread has other locks acquired?  If so, I'll drop the second commit that moves this.

Reply:
I've realized in hindsight that this question is rather stupid, as "background threads" acquiring `pending_background_events` is fine even if other threads are holding other locks at the same time. We just need to ensure that every single thread acquires the locks according to the lock order to avoid deadlocks.